components/datasets/form/steps/Step1.js

```diff
@@ -57,6 +64,42 @@ class Step1 extends React.Component {
     this.props.onChange({ legend });
   }
 
+  onSubscribableChange(obj) {
+    const { subscribable } = this.props.form;
+    const newSubscribable = subscribable.slice(0);
+    if (obj.type) {
+      newSubscribable[obj.id].type = obj.type;
+    } else if(obj.value) {
+      newSubscribable[obj.id].value = obj.value;
+    }
+    this.props.onChange({ subscribable: newSubscribable });
+  }
+
+  onSubscribableCheckboxChange(checked) {
+    this.setState({
+      subscribableSelected: checked,
+    });
+    if (checked) {
+      this.props.onChange({ subscribable: [{ type: '', value:'', id: 0 }] });
+    } else {
+      this.props.onChange({ subscribable: [] });
+    }
+  }
+
+  handleRemoveSubscription(index) {
+    const { subscribable } = this.props.form;
```

Comment:
The same here. You are assuming that the index would be the id. This is not true...

Reply:
```
  handleRemoveSubscription(id) {
    const subscribable = this.props.form.subscribable.slice(0);
    const index = subscribable.findIndex(s => s.id === id);

    subscribable.splice(index, 1)
    this.props.onChange({ subscribable });
  }

```

Reply:
👍